tests/run/t3346e.flags

```diff
@@ -0,0 +1 @@
+-language:Scala2
```

Comment:
This won't work, flags files are ignored. For pos tests we have a special `pos-scala2` folder so we would need a `run-scala2` folder here and correct configuration in the tests, it seems easier to change the test to not use Scala2-only features. Why did you enable this test in particular?